File: JavaJanken/app/src/main/java/com/example/javajanken/MainActivity.java
package com.example.javajanken;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    ImageView cpu;
    ImageView player;
    TextView result;
    Random r = new Random();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        cpu = findViewById(R.id.cpu);
        player = findViewById(R.id.player);
        result = findViewById(R.id.result);

    }

    public void goo (View v){
        player.setImageResource(R.drawable.goo);
        int random = r.nextInt(3);

        if (random == 0){
            cpu.setImageResource(R.drawable.goo);
            result.setText("引き分け");
        }else if (random == 1){
            cpu.setImageResource(R.drawable.choki);
            result.setText("勝ち");
        }else if (random == 2){
            cpu.setImageResource(R.drawable.paa);
            result.setText("負け");
        }
    }

    public void choki (View v){
        player.setImageResource(R.drawable.choki);
        int random = r.nextInt(3);

        if (random == 0){
            cpu.setImageResource(R.drawable.choki);
            result.setText("引き分け");
        }else if (random == 1){
            cpu.setImageResource(R.drawable.paa);
            result.setText("勝ち");
        }else if (random == 2){
            cpu.setImageResource(R.drawable.goo);
            result.setText("負け");
        }
    }

    public void paa (View v){
        player.setImageResource(R.drawable.paa);
        int random = r.nextInt(3);

        if (random == 0){
            cpu.setImageResource(R.drawable.paa);
            result.setText("引き分け");
        }else if (random == 1){
            cpu.setImageResource(R.drawable.goo);
            result.setText("勝ち");
        }else if (random == 2){
            cpu.setImageResource(R.drawable.choki);
            result.setText("負け");
        }
    }
}

File: Janken/app/src/main/java/com/example/janken/MainActivity.kt
package com.example.janken

import androidx.appcompat.app.AppCompatActivity

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import java.util.Random

class MainActivity : AppCompatActivity() {

    internal lateinit var cpu: ImageView
    internal lateinit var player: ImageView
    internal lateinit var result: TextView
    internal var r = Random()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        cpu = findViewById(R.id.cpu)
        player = findViewById(R.id.player)
        result = findViewById(R.id.result)
    }

    fun goo(v: View) {
        player.setImageResource(R.drawable.goo)
        val random = r.nextInt(3)

        if (random == 0) {
            cpu.setImageResource(R.drawable.goo)
            result.text = "引き分け"
        } else if (random == 1) {
            cpu.setImageResource(R.drawable.choki)
            result.text = "勝ち"
        } else if (random == 2) {
            cpu.setImageResource(R.drawable.paa)
            result.text = "負け"
        }
    }

    fun choki(v: View) {
        player.setImageResource(R.drawable.choki)
        val random = r.nextInt(3)

        if (random == 0) {
            cpu.setImageResource(R.drawable.choki)
            result.text = "引き分け"
        } else if (random == 1) {
            cpu.setImageResource(R.drawable.paa)
            result.text = "勝ち"
        } else if (random == 2) {
            cpu.setImageResource(R.drawable.goo)
            result.text = "負け"
        }
    }

    fun paa(v: View) {
        player.setImageResource(R.drawable.paa)
        val random = r.nextInt(3)

        if (random == 0) {
            cpu.setImageResource(R.drawable.paa)
            result.text = "引き分け"
        } else if (random == 1) {
            cpu.setImageResource(R.drawable.goo)
            result.text = "勝ち"
        } else if (random == 2) {
            cpu.setImageResource(R.drawable.choki)
            result.text = "負け"
        }
    }
}
